### DISCUSSION TITLE
Simplify expressions involving Levi-Civita Tensor contractions

### DISCUSSION THREAD

**Discussion by susobhan21** [FeynCalc/feyncalc#252] [Q&A]
Hello FeynCalc team and FeynCalc users!.

I can simplify the contractions of four-vectors by explicitly defining the corresponding scalar products. But I am unable to simplify expressions involving contractions of Levi-Civita tensor with 4 four-vectors. Can someone please help me on how to simplify such expressions?

For Example,
![image](https://github.com/FeynCalc/feyncalc/assets/84768504/d8d70bfa-9fe6-48ec-a406-b41fd135e768)

could be simplified to 

![image](https://github.com/FeynCalc/feyncalc/assets/84768504/00b96ea2-6628-4b45-81f3-1735aaa75dd7)

by explicitly defining the scalar products of momenta and polarization vectors and then contracting the FCFAConvert amplitude. But, still the Levi-Civita contractions remain unsimplified. I have tried to explicitly define the Levi-Civita contractions with other vectors using the Contract[....] function, that didn't work out. When defining by the Contract[..] function, I can obtain the contraction result for say 
Contract[LC[\[Mu], \[Nu], \[Rho], \[Sigma]] FV[k1, \[Mu]] FV[
   k2, \[Nu]] Polarization[k1, -I] Polarization[p2, I]] 
in the correct simplified form, but the Contract[FCFAAmplitude[..]] didn't work and still shows unsimplified expressions of the Levi-Civita contractions....

I am using FeynCal 10.0.0 (dev. version) and Mathematica 12.1.
Hope you can help me. Thank you.

**Comment by vsht** [comment]
Hi,

what do you mean by simplifying Levi-Civita contractions with other vectors? How would you write something simpler than that?

**Reply by susobhan21** [reply]
I meant by explicitly substituting them for a particular kinematic configuration(for example,say, (for ++  ---> ++ scattering amplitude)
 as I did for the polarizations and the four-vector scalar products. I had defined,
![image](https://github.com/FeynCalc/feyncalc/assets/84768504/967b50fa-374c-4d7a-aae9-7f2dcc00912b)
 and then defined the scalar products of each possible four-vector and/or polarization scalar product by using the above shown vectors and the dotfunc. By defining such scalar products, simplified the expression as mentioned above(from Fig-1 in previous post to Fig. 2 of precious post) (all scalar products involving four-vectors and/or polarizations have been substituted and simplified in terms of "p" and "theta"). 
I want something similar with the Levi-Civita tensor contracting with four-vectors and polarizations by defining these kind of contractions explicitly in terms of "p" and "\theta" and then finally simplify the whole amplitude in terms of "p" and "\theta".

Something like this:
![image](https://github.com/FeynCalc/feyncalc/assets/84768504/1b4e0fd3-69fe-4171-861d-145aef31a4d4)
(for ++  ---> ++ scattering amplitude)
But this does not simplify the Contract[FCFAConvert[...]] in terms of only "p" and "\theta"

**Reply by vsht** [reply]
Substituting explicit components is not something internally supported in FeynCalc. You can look at the FCI form
of those Levi-Civitas and define replacement rules for that, but there's certainly no built-in function doing all that for you.
